src/server/game/Loot/LootMgr.h

```diff
@@ -115,19 +115,20 @@ struct Loot;
 struct LootStoreItem
 {
     uint32  itemid;                                         // id of the item
-    float   chance;                                         // always positive, chance to drop for both quest and non-quest items, chance to be used for refs
-    int32   mincountOrRef;                                  // mincount for drop items (positive) or minus referenced TemplateleId (negative)
+    uint32  reference;                                      // referenced TemplateleId
+    float   chance;                                         // chance to drop for both quest and non-quest items, chance to be used for refs
+    bool    needs_quest : 1;                                // quest drop (quest is required for item to drop)
     uint16  lootmode;
-    uint8   group       :7;
-    bool    needs_quest :1;                                 // quest drop (negative ChanceOrQuestChance in DB)
-    uint8   maxcount    :8;                                 // max drop count for the item (mincountOrRef positive) or Ref multiplicator (mincountOrRef negative)
+    uint8   groupid     : 7;
+    uint8   mincount;                                       // mincount for drop items
+    uint8   maxcount;                                       // max drop count for the item mincount or Ref multiplicator
     ConditionList conditions;                               // additional loot condition
 
-    // Constructor, converting ChanceOrQuestChance -> (chance, needs_quest)
+    // Constructor
     // displayid is filled in IsValid() which must be called after
-    LootStoreItem(uint32 _itemid, float _chanceOrQuestChance, uint16 _lootmode, uint8 _group, int32 _mincountOrRef, uint8 _maxcount)
-        : itemid(_itemid), chance(fabs(_chanceOrQuestChance)), mincountOrRef(_mincountOrRef), lootmode(_lootmode),
-        group(_group), needs_quest(_chanceOrQuestChance < 0), maxcount(_maxcount)
+    LootStoreItem(uint32 _itemid, uint32 _reference, float _chance, bool _needs_quest, uint16 _lootmode, uint8 _groupid, int32 _mincount, uint8 _maxcount)
+        : itemid(_itemid), reference(_reference), chance(_chance), lootmode(_lootmode),
```

Comment:
Please, re-order the initialization. Put loot_mode after needs_quest to fix travis ;)
```suggestion
        : itemid(_itemid), reference(_reference), chance(_chance), needs_quest(_needs_quest), lootmode(_lootmode),
```